packages/ckeditor5-list/docs/_snippets/features/lists-document.html

```diff
@@ -0,0 +1,122 @@
+<div id="snippet-lists-document">
+	<h2>Top five places to visit this summer</h2>
+	<h3>...and stay on the budget.</h3>
+	<ol>
+		<li>
+			<h4>Gran Canaria, Spain</h4>
+			<figure class="image image_resized image-style-align-right" style="width:30%;">
+				<img src="%BASE_PATH%/assets/img/gran-canaria.jpg">
+				<figcaption>Ocean shore of Gran Canaria</figcaption>
+			</figure>
```

Comment:
Confirmed, can recreate.

![](https://user-images.githubusercontent.com/68123541/191261934-21295b29-69d0-4be5-a5f2-a63a1f955453.gif)

@oleq can you please have a look at it?

Reply:
What is expected result here? (I'm trying to figure out if this is a problem with the demo, with the `DocumentList` feature, or not a problem at all)

Reply:
The expected result is that you can backspace the "2" from the list without deleting the image.

Reply:
OK now I get it. 

Well, the problem with HTML is that it's HTML 😛. And the browser interacts with it in a certain way. In the example, the `<figure>` with the image is at the beginning of the list item but it has a `float` CSS property.

[https://github.com/ckeditor/ckeditor5/pull/12489/files#diff-ad7287302531b66dd41a68b6f5fe149defa9a047e6083d1f4d3fdae3bb640193R7-R10](https://github.com/ckeditor/ckeditor5/pull/12489/files#diff-ad7287302531b66dd41a68b6f5fe149defa9a047e6083d1f4d3fdae3bb640193R7-R10)

If you put a caret at the beginning of the list item, the web browser still remembers the image is anchored there despite being displayed somewhere else and the interaction works in a specific way (screencast).

We cannot do much about it. If we ignored the image in that case (because technically we could), the user would lose the way to remove the image using the keyboard. 

I suppose we could move the `<figure>` after the paragraph to work around this but it will result in similar problems with `Delete` at the end of the list item.

Reply:
Is it possible to merge those list items by pressing backspace when the upper fake caret is active?

Reply:
That's all I can come up with.

![](https://user-images.githubusercontent.com/68123541/191722282-6399bc0c-91c1-4218-b6ec-c064ef86d98a.gif)

And, by the way, the toolbar doesn't stick below the topbar:

![](https://user-images.githubusercontent.com/68123541/191722370-e5f56826-b05a-4ae7-a819-eab9429d2c05.gif)

Reply:
So probably there is a missing integration with WTA fake caret. Could you create an issue for that?

Reply:
#12502

Reply:
It is good to go now I think. We could extend the content later after fixing the glitches.